platform/v1alpha1/types.go

```diff
@@ -0,0 +1,73 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// PlatformOperatorSpec defines the desired state of PlatformOperator.
+type PlatformOperatorSpec struct {
+	// PackageName contains the desired OLM-based Operator package name
+	// that is defined in an existing CatalogSource resource in the cluster.
+	//
+	// This configured package will be managed with the cluster's lifecycle. In
+	// the current implementation, it will be retrieving this name from a list of
+	// supported operators out of the redhat-operators catalog included with
+	// OpenShift.
+	// +kubebuilder:validation:Required
+	PackageName string `json:"packageName"`
```

Comment:
I had a comment before about API evolution and making this a struct, did we ever conclude on that?

```
package:
  name: foo
```
This would be easier to add additional fields to in the future if you need to add additional configuration around packages

Reply:
Good call, Joel. Adding that now.